### PR TITLE
APPDRUSTI-16

### PR DESCRIPTION
**Pull request by BOSSHK-SPEC**

### CODE REVIEW

File: lib/CreateStaff.dart [hunk 1]
```diff
@@ -1,16 +1,935 @@
...
+                    shape: RoundedRectangleBorder(
+                        borderRadius:
+                            BorderRadius.circular(12.0)), //this right here
+                    child: Container(
+                      height: 300.0,
+                      width: 300.0,
+                      child: Column(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        children: <Widget>[
+                          Padding(
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:122]: remove cool and awesome

File: lib/CreateStaff.dart [hunk 2]
```diff
@@ -1,16 +1,935 @@
...
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(title: Text('Create Staff')),
+        appBar: AppBar(
+          title: new Text("Create Staff"),
+          actions: <Widget>[
+            IconButton(
+                icon: Icon(
+                  Icons.unarchive,
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:108]: change to upload button or uparrow

File: lib/CreateStaff.dart [hunk 3]
```diff
@@ -1,16 +1,935 @@
...
+                              style: TextStyle(color: Colors.red),
+                            ),
+                          ),
+                          Padding(padding: EdgeInsets.only(top: 50.0)),
+                          TextButton(
+                              onPressed: () {
+                                Navigator.of(context).pop();
+                              },
+                              child: Text(
+                                'Got It!',
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:142]: remove got it text field in container

File: lib/CreateStaff.dart [hunk 4]
```diff
@@ -1,16 +1,935 @@
...
+                              focusedBorder: OutlineInputBorder(
+                                borderSide:
+                                    BorderSide(color: Colors.black, width: 1.0),
+                              ),
+                              enabledBorder: OutlineInputBorder(
+                                borderSide:
+                                    BorderSide(color: Colors.black, width: 1.0),
+                              ),
+                              hintText: 'First Name',
+                              labelText: "F- name",
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:237]: change label to firstName

File: lib/CreateStaff.dart [hunk 5]
```diff
@@ -1,16 +1,935 @@
...
+                            ),
+                            validator: (val) {
+                              if (val == null || val.isEmpty) {
+                                return 'Please enter F-Name';
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:241]: change to First name instead of F-Name everywhere

File: lib/CreateStaff.dart [hunk 6]
```diff
@@ -1,16 +1,935 @@
...
+                                borderSide:
+                                    BorderSide(color: Colors.black, width: 1.0),
+                              ),
+                              hintText: 'Middle Name',
+                              labelText: "M-Name",
+                            ),
+                            validator: (val) {
+                              if (val == null || val.isEmpty) {
+                                return 'Please enter M-Name';
+                              }
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:268]: change M- Name to middle name everywhere

File: lib/CreateStaff.dart [hunk 7]
```diff
@@ -1,16 +1,935 @@
...
+                                  vertical: 10.0, horizontal: 10.0),
+                              focusedBorder: OutlineInputBorder(
+                                borderSide:
+                                    BorderSide(color: Colors.black, width: 1.0),
+                              ),
+                              enabledBorder: OutlineInputBorder(
+                                borderSide:
+                                    BorderSide(color: Colors.black, width: 1.0),
+                              ),
+                              hintText: 'Last Name',
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:288]: change L-name to last name

File: lib/CreateStaff.dart [hunk 8]
```diff
@@ -1,16 +1,935 @@
...
+                          groupValue: val,
+                          onChanged: (value) {
+                            setState(() {
+                              val = 2;
+                            });
+                          },
+                          activeColor: Colors.black,
+                        ),
+                      ),
+                      if (val == 1)
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:397]: remove hi gentelman and ladies

File: lib/CreateStaff.dart [hunk 9]
```diff
@@ -1,16 +1,935 @@
...
+                        Text('hii gentelman')
+                      else
+                        (Text('hii lady')),
+                      ElevatedButton(
+                        onPressed: () {
+                          if (_formKey.currentState!.validate()) {}
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:403]: add global error message here

File: lib/CreateStaff.dart [hunk 10]
```diff
@@ -1,16 +1,935 @@
...
+                                  ),
+                                  enabledBorder: OutlineInputBorder(
+                                    borderSide: BorderSide(
+                                        color: Colors.black, width: 1.0),
+                                  ),
+                                  hintText: 'Phone Number-2',
+                                  labelText: "Phone Number-2"),
+                              validator: (val) {
+                                if (val == null || val.isEmpty) {
+                                  return 'Please enter registration number';
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:503]: why the error message is please enter registration number?

File: lib/CreateStaff.dart [hunk 11]
```diff
@@ -1,16 +1,935 @@
...
+                              onChanged: (value) {
+                                RegExp exp =
+                                    RegExp(r'(^(?:[+0]9)?[0-9]{10,12}$)');
+                                if (!exp.hasMatch(value)) {
+                                  setState(() {
+                                    validPhoneNo = false;
+                                  });
+                                } else {
+                                  setState(() {
+                                    validPhoneNo = true;
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:483]: no need of phone 2 validation

File: lib/CreateStaff.dart [hunk 12]
```diff
@@ -1,16 +1,935 @@
...
+                                  isDense: true,
+                                  contentPadding: new EdgeInsets.symmetric(
+                                      vertical: 10.0, horizontal: 10.0),
+                                  focusedBorder: OutlineInputBorder(
+                                    borderSide: BorderSide(
+                                        color: Colors.black, width: 1.0),
+                                  ),
+                                  enabledBorder: OutlineInputBorder(
+                                    borderSide: BorderSide(
+                                        color: Colors.black, width: 1.0),
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:556]: no need of email2 validation

File: lib/CreateStaff.dart [hunk 13]
```diff
@@ -1,16 +1,935 @@
...
+                                    validEmail = true;
+                                  } else {
+                                    validEmail = false;
+                                  }
+                                }),
+                              },
+                            )),
+                        ElevatedButton(
+                          onPressed: () {
+                            if (_formKey.currentState!.validate()) {}
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:577]: add global error message

File: lib/CreateStaff.dart [hunk 14]
```diff
@@ -1,16 +1,935 @@
...
+                            onChanged: (value) {
+                              setState(() {
+                                val = 2;
+                              });
+                            },
+                            activeColor: Colors.black,
+                          ),
+                        ),
+                        CheckboxListTile(
+                          title: Text("Both address are Same"),
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:631]: Both addresses it should be, also add current 'Address'

File: lib/CreateStaff.dart [hunk 15]
```diff
@@ -1,16 +1,935 @@
...
+                          controlAffinity: ListTileControlAffinity.leading,
+                          value: address_Value,
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:633]: Add all expansiontiles to a method and use it everywhere

File: lib/CreateStaff.dart [hunk 16]
```diff
@@ -1,16 +1,935 @@
...
+                        ),
+                        Container(
+                          child: val == 1
+                              ? createAddressForm(true)
+                              : createAddressForm(false),
+                        ),
+                        ElevatedButton(
+                          onPressed: () {
+                            if (_formKey.currentState!.validate()) {}
+                            // Validate returns true if the form is valid, or false otherwise.
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:654]: Add global error message

File: lib/CreateStaff.dart [hunk 17]
```diff
@@ -1,16 +1,935 @@
...
+                          },
+                          child: Text('Save'),
+                        ),
+                      ]),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    ElevatedButton(
+                      onPressed: () {
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:664]: please add functionality to back button

File: lib/CreateStaff.dart [hunk 18]
```diff
@@ -1,16 +1,935 @@
...
+                ),
+                enabledBorder: OutlineInputBorder(
+                  borderSide: BorderSide(color: Colors.black, width: 1.0),
+                ),
+                hintText: 'Addressline 1',
+                labelText: 'Addressline 1',
+              ),
+              validator: (val) {
+                if (val == null || val.isEmpty) {
+                  return 'Please enter Address';
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:709]: change to addressline 1

File: lib/CreateStaff.dart [hunk 19]
```diff
@@ -1,16 +1,935 @@
...
+                focusedBorder: OutlineInputBorder(
+                  borderSide: BorderSide(color: Colors.black, width: 1.0),
+                ),
+                enabledBorder: OutlineInputBorder(
+                  borderSide: BorderSide(color: Colors.black, width: 1.0),
+                ),
+                hintText: 'Addressline 2',
+                labelText: 'Addressline 2',
+              ),
+              validator: (val) {
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:734]: addressline 2 can be empty, so no need of this validation

File: lib/CreateStaff.dart [hunk 20]
```diff
@@ -1,16 +1,935 @@
...
+                }
+              },
+            ),
+          ),
+          Padding(
+              padding:
+                  EdgeInsets.only(bottom: 6.0, left: 60, right: 60, top: 10),
+              child: DropdownButtonFormField(
+                decoration: new InputDecoration(
+                    isDense: true,
```
Comment (sahana-drusti) [inline lib/CreateStaff.dart:746]: add validation to state, district, taluk as we have select in the List

File: lib/CreateStudent.dart [hunk 21]
```diff
@@ -1,135 +1,1160 @@
...
+                            onChanged: (value) {
+                              setState(() {
+                                religionValue = value.toString();
+                              });
+                            },
+                          )),
+                      ElevatedButton(
+                        onPressed: () {
+                          if (_formKey.currentState!.validate()) {}
+                          // Validate returns true if the form is valid, or false otherwise.
```
Comment (sahana-drusti) [inline lib/CreateStudent.dart:626]: add global error message

File: lib/CreateStudent.dart [hunk 22]
```diff
@@ -1,135 +1,1160 @@
...
+                                    validEmail = true;
+                                  } else {
+                                    validEmail = false;
+                                  }
+                                }),
+                              },
+                            )),
+                        Padding(
+                            padding: EdgeInsets.only(
+                                bottom: 6.0, left: 60, right: 60, top: 10),
```
Comment (sahana-drusti) [inline lib/CreateStudent.dart:766]: same changes required for student page as well

File: lib/CreateStudent.dart [hunk 23]
```diff
@@ -1,135 +1,1160 @@
...
+      date = newDate;
+      dateController.value=TextEditingValue(text: getDate());
+    });
   }
 }
+
+
+
+
+
```
Comment (sahana-drusti) [inline lib/CreateStudent.dart:1160]: remove picker.dart file as it is not used anywhere

File: lib/CreateStaff.dart (hunk 1 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:122]: done

File: lib/CreateStaff.dart (hunk 2 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:108]: ok

File: lib/CreateStaff.dart (hunk 3 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:142]: ok

File: lib/CreateStaff.dart (hunk 4 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:237]: ok

File: lib/CreateStaff.dart (hunk 5 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:241]: ok

File: lib/CreateStaff.dart (hunk 6 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:268]: ok

File: lib/CreateStaff.dart (hunk 7 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:288]: ok

File: lib/CreateStaff.dart (hunk 8 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:397]: ok

File: lib/CreateStaff.dart (hunk 10 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:503]: ok

File: lib/CreateStaff.dart (hunk 11 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:483]: done

File: lib/CreateStaff.dart (hunk 12 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:556]: ok

File: lib/CreateStaff.dart (hunk 17 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:664]: ok

File: lib/CreateStaff.dart (hunk 18 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:709]: ok

File: lib/CreateStaff.dart (hunk 19 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:734]: ok

File: lib/CreateStudent.dart (hunk 22 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStudent.dart:766]: ok

File: lib/CreateStudent.dart (hunk 23 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStudent.dart:1160]: ok

File: lib/CreateStaff.dart (hunk 15 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:633]: ok

File: lib/CreateStaff.dart (hunk 14 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:631]: ok

File: lib/CreateStaff.dart (hunk 20 above)
Comment (BOSSHK-SPEC) [reply lib/CreateStaff.dart:746]: done

**Pull request merged**